### PR TITLE
values.js - implement toString/inspect spec

### PR DESCRIPTION
**Pull request by bkutil**
Change the output of toString and inspect for Juttle values to conform
to the specification outlined here:

https://github.com/juttle/juttle/wiki/String-Representation-of-Juttle-Values

The spec is encoded into a set of accompanying tests (taken from the
specfication table).

### CODE REVIEW

**Comment by bkutil** [comment]
cc: @dmajda

File: test/runtime/values.spec.js [hunk 1]
```diff
@@ -133,4 +134,175 @@ describe('Values tests', function () {
...
+                {
+                    value: '',
+                    expected: ''
+                },
+                {
+                    value: 'abcd',
+                    expected: 'abcd'
+                },
+                {
+                    value: new RegExp(),
```
Comment (dmajda) [inline test/runtime/values.spec.js]: I’d be slightly more explicit and do `RegExp('')`.

File: test/runtime/values.spec.js [hunk 2]
```diff
@@ -133,4 +134,175 @@ describe('Values tests', function () {
...
+                {
+                    value: new JuttleMoment.duration('5', 'seconds'),
+                    expected: '00:00:05.000'
+                },
+                {
+                    value: new JuttleMoment('2015-01-01T00:00:05'),
+                    expected: '2015-01-01T00:00:05.000Z'
+                },
+                {
+                    value: new Filter({}, 'a < 5'),
```
Comment (dmajda) [inline test/runtime/values.spec.js]: I’d slightly prefer to pass the AST too (see e.g. [here](https://github.com/juttle/juttle/blob/master/test/compiler/inputs.spec.js#L92-L103)).

File: test/runtime/values.spec.js [hunk 3]
```diff
@@ -133,4 +134,175 @@ describe('Values tests', function () {
             });
         });
     });
+
+    describe('string representation', function() {
+        it('toString', function() {
```
Comment (dmajda) [inline test/runtime/values.spec.js]: This `describe`/`it` combination is weird. What about something like this:

```javascript
describe('toString', function() {
    it('returns correct string representation', function() {
        // ...
```

**Comment by dmajda** [comment]
All notes for `toString` tests also apply to `inspect`.

+1 with those addressed and tests fixed

**Comment by bkutil** [comment]
Fixed tests, added the AST to filter, renamed tests.

**Pull request merged**